apps/web/src/components/layout/components/SideNav.tsx

```diff
@@ -47,12 +46,16 @@ export function SideNav({}: Props) {
     }
 
     setIsLoading(true);
-
     const tokenResponse = await api.post(`/v1/auth/environments/${targetEnvironment?._id}/switch`, {});
+
+    if (!tokenResponse.token) {
+      setIsLoading(false);
+
+      return;
+    }
+
     setToken(tokenResponse.token);
     setIsLoading(false);
-
-    await queryClient.refetchQueries();
```

Comment:
I see that it was removed, why? The UI will contain data from old environment of not prefetched. OR am I missing something?

Reply:
NVM, seen you did another approach.